packages/mobile/src/storage/keychain.tsx

```diff
@@ -19,6 +19,15 @@ export async function storeItem({ key, value }: SecureStorage) {
     if (result === false) {
       throw new Error('Store result false')
     }
+
+    // check that we can correctly read the keychain before proceeding
+    const retrievedResult = await retrieveStoredItem(key)
+    if (retrievedResult !== value) {
+      await removeStoredItem(key)
+      Logger.error(`${TAG}@storeItem`, 'Retrieved value does not match stored value')
+      throw new Error('Retrieved value does not match stored value')
+    }
+
```

Comment:
i don't think it will conflict with the biometry changes - the store pin with biometry function has the same logic to read back what was stored in the consumer of this method, so that won't be needed anymore with this change :)